File: apps/ResidentApp.Android/app/src/main/java/com/acls/resident/data/remote/dto/AuthDtos.kt
package com.acls.resident.data.remote.dto

import com.google.gson.annotations.SerializedName

data class RegisterResidentRequestDto(
    @SerializedName("invitationToken") val invitationToken: String,
    @SerializedName("email") val email: String,
    @SerializedName("password") val password: String,
    @SerializedName("firstName") val firstName: String,
    @SerializedName("lastName") val lastName: String,
    @SerializedName("phone") val phone: String? = null
)

data class LoginRequestDto(
    @SerializedName("email") val email: String,
    @SerializedName("password") val password: String
)

data class AuthTokenResponseDto(
    @SerializedName("accessToken") val accessToken: String,
    @SerializedName("expiresAt") val expiresAt: String,
    @SerializedName("userId") val userId: Int,
    @SerializedName("role") val role: String,
    @SerializedName("propertyId") val propertyId: Int
)

File: apps/ResidentApp.iOS/Sources/AuthDTOs.swift
import Foundation

// MARK: - Auth Request DTOs

struct LoginRequest: Encodable {
    let email: String
    let password: String
}

struct RegisterResidentRequest: Encodable {
    let invitationToken: String
    let email: String
    let password: String
    let firstName: String
    let lastName: String
    let phone: String?
}

// MARK: - Auth Response DTOs

struct AuthTokenResponse: Decodable {
    let accessToken: String
    let expiresAt: Date
    let userId: Int
    let role: String
    let propertyId: Int
}
